pgxn/neon/walproposer_pg.c

```diff
@@ -203,6 +205,16 @@ nwp_register_gucs(void)
 												 * GUC_LIST_QUOTE */
 							   NULL, assign_neon_safekeepers, NULL);
 
+	DefineCustomStringVariable(
+							   "neon.safekeeper_extra_conninfo_options",
+							   "libpq keyword parameters and values to apply to safekeeper connections",
+							   NULL,
+							   &safekeeper_extra_conninfo_options,
+							   "",
+							   PGC_SIGHUP,
```

Comment:
I think `PGC_POSTMASTER` fits better here, because we don't have code to reconnect to safekeepers when this option changes right now.
Or need to implement `assign_neon_safekeeper_extra_conninfo_options` hook to restart the walproposer process if the option changes similarly to `assign_neon_safekeepers`

Reply:
Thanks for catching this. I agree with you.